zipline/src/engineMain/kotlin/app/cash/zipline/internal/SourceMapParser.kt

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (C) 2021 Square, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package app.cash.zipline.internal
+
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import okio.Buffer
+import okio.BufferedSource
+import okio.IOException
+
+/**
+ * Parses the contents of a source map file to enable mapping from the elements in the transformed
+ * (e.g. minified) source to the elements in the original source. This is an implementation of the
+ * [Source Map Revision 3 Proposal](https://sourcemaps.info/spec.html).
+ */
+internal class SourceMapParser {
```

Comment:
Seems like we can drop the class, make all the data classes top-level and internal, and make `parse` exist directly on its `companion object`. This class is not stateful and does nothing except perform namespace scoping.

Reply:
Agree, done: b1120ab